SimG4CMS/Calo/src/HCalSD.cc

```diff
@@ -627,6 +632,7 @@ uint32_t HCalSD::setDetUnitId (int det, const G4ThreeVector& pos, int depth, int
 uint32_t HCalSD::setDetUnitId (HcalNumberingFromDDD::HcalID& tmp) { 
   modifyDepth(tmp);
   uint32_t id = (numberingScheme) ? numberingScheme->getUnitID(tmp) : 0;
+  if ((!testNumber) && hcalTestNS_) hcalTestNS_->compare(tmp,id);
```

Comment:
testNumber  is a member variable and set at the beginning of the class through a parameter statement